### PR TITLE
#162192931 set up react project for my diary

### PR DESCRIPTION
**Pull request by kakaemma**
#### What does this PR do?
This PR creates a react application for mydiary,
Adds redux to the application,
Set's up react router and also adds a landing page with a login form.

#### How should this be manually tested?
Clone the application with `git clone `
install application dependencies with `npm install`
Run the application with `npm start`
On success, a landing page with a login form should will be displayed.

#### What are the relevant pivotal tracker stories?
[#162192931](https://www.pivotaltracker.com/story/show/162192931)

#### Any background context you want to add?
N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+<<<<<<< HEAD
```
Comment (MuhanguziDavid) [inline .gitignore]: @kakaemma , looks like you have conflicts from line 24-29

Comment (kakaemma) [reply .gitignore]: Thank you @MuhanguziDavid, I have worked on the feedback.

**Review by MuhanguziDavid — COMMENTED** [review]
Nice work @kakaemma , please consider setting up a testing environment. You could use `enzyme`

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <!--
+      manifest.json provides metadata used when your web app is added to the
+      homescreen on Android. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css" />
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json">
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
```
Comment (samuelbwambale) [inline public/index.html]: @kakaemma Consider deleting these comments

**Comment by samuelbwambale** [comment]
@kakaemma I have not seen any test files. Could it be because this is the boiler-plate of the project?

**Comment by kakaemma** [comment]
@samuelbwambale , the test files will be in a chore for setting up the testing environment

File: public/index.html (hunk 2 above)
Comment (kakaemma) [reply public/index.html]: Thank you @samuelbwambale , I have implemented the feedback

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Review by samuelbwambale — APPROVED** [review]
LGTM

**Pull request merged**